plans/objchange/compatible.go

```diff
@@ -73,9 +72,8 @@ func assertObjectCompatible(schema *configschema.Block, planned, actual cty.Valu
 		}
 	}
 	for name, blockS := range schema.BlockTypes {
-		// Unmark values before testing compatibility
-		plannedV, _ := planned.GetAttr(name).UnmarkDeep()
-		actualV, _ := actual.GetAttr(name).UnmarkDeep()
+		plannedV, _ := planned.GetAttr(name).Unmark()
```

Comment:
The purpose of `AssertObjectCompatible` is to assess the validity of values (planned vs. actual) and against the schema. For blocks, we only need to unmark at the block level (not `Deep`) because further recursive layers get to the Attributes level where we also do unmarking.